src/Card.cpp

```diff
@@ -1,22 +1,46 @@
 #include "Card.h"
+#include "ESPDash.h"
+#include "Tab.h"
+
+// Integral type to string pairs events
+// ID, type
+struct WidgetNames cardTags[] = {
+  {GENERIC_CARD, "generic"},
+  {TEMPERATURE_CARD, "temperature"},
+  {HUMIDITY_CARD, "humidity"},
+  {STATUS_CARD, "status"},
+  {SLIDER_CARD, "slider"},
+  {BUTTON_CARD, "button"},
+  {PROGRESS_CARD, "progress"},
+};
 
 /*
   Constructor
 */
-Card::Card(ESPDash *dashboard, const int type, const char* name, const char* symbol, const int min, const int max){
+Card::Card(ESPDash *dashboard, const int type, const char* name, const char* symbol, const int min, const int max):
+Widget()
+{
   _dashboard = dashboard;
-  #if defined(ESP8266)
-    _id = RANDOM_REG32;
-  #elif defined(ESP32)
-    _id = esp_random();
-  #endif
+  _tab = nullptr;
   _type = type;
   _name = name;
   _symbol = symbol;
   _value_min = min;
   _value_max = max;
   _dashboard->add(this);
 }
+Card::Card(Tab *tab, const int type, const char* name, const char* symbol, const int min, const int max):
```

Comment:
Maybe the pointer could be of type etl::variant<ESPDash*, Tab*> and reuse the code.

Reply:
Changes the constructor, hence will not do